src/vs/workbench/parts/files/electron-browser/fileActions.ts

```diff
@@ -1392,6 +1385,33 @@ export function validateFileName(parent: ExplorerItem, name: string, allowOverwr
 	return null;
 }
 
+
+interface IMappedPath {
+	fullPathAlreadyExists: boolean;
+	lastExistingPathSegment: {
+		isFile: boolean;
+		name: string;
+	};
+}
+
+function mapPathsToExistingFolders(parent: ExplorerItem, pathNames: string[]): IMappedPath {
```

Comment:
Not very happy about the names here. When I read "map" I expect something like an actual map.
I would rename to something simplier, maybe something with "check" and "path". I leave this up to you